pytorch_lightning/callbacks/pruning.py

```diff
@@ -0,0 +1,336 @@
+                    parameters_to_prune.append((m, p))
+
+    if isinstance(parameters_to_prune, (tuple, list)) \
+       and len(parameters_to_prune) > 0 and not is_parameters_to_prune_none:
+
+        if all(isinstance(p, (list, tuple)) and _check_parameters_to_prune(p) for p in parameters_to_prune):
+
+            missing_modules = []
+            missing_parameters = []
+
+            for module, param_name in parameters_to_prune:
+                if module not in current_modules:
+                    missing_modules.append(module)
+                    continue
+
+                parameter = getattr(module, param_name)
+
+                if parameter is None:
+                    missing_parameters.append(parameter)
+
+            if len(missing_modules) > 0 or len(missing_parameters) > 0:
+                raise MisconfigurationException(
+                    "Ths provided parameters_to_tune doesn't exist in the model. "
+                    f"Found mismatching modules: {missing_modules} and missing_parameters: {missing_parameters}"
+                )
+
+        else:
+            raise MisconfigurationException(
+                "The provided parameters_to_prune should either be list of tuple "
+                "with 2 elements: (nn.Module in your model, parameter_name_to_prune) or None")
+    else:
+        if not isinstance(parameters_to_prune, (list, tuple)):
+            raise MisconfigurationException(
+                "The provided parameters_to_prune should either be list of tuple "
+                "with 2 elements: (nn.Module in your model, parameter_name_to_prune) or None")
+
+    return parameters_to_prune
+
+
+class ModelPruning(Callback):
+
+    PARAMETER_NAMES = ["weight", "bias"]
+
+    def __init__(
+        self,
+        pruning_fn: Union[Callable, str],
+        parameters_to_prune: Optional[List[Tuple[nn.Module, str]]] = None,
+        parameter_names: List[str] = ["weight"],
+        use_global_unstructured: bool = True,
+        amount: Optional[Union[int, float]] = 0.5,
+        prune_on_epoch_end: Optional[bool] = False,
+        prune_on_fit_end: Optional[bool] = True,
+        make_pruning_permanent: Optional[bool] = True,
+        use_lottery_ticket_hypothesis: Optional[bool] = True,
+        pruning_dim: Optional[int] = None,
+        pruning_norm: Optional[int] = None,
+    ) -> None:
+        """
+
+        Pruning Callback relying on PyTorch prune utils.
+
+        This callback is responsible to prune networks parameters
+        during your training.
+
+        Find here the PyTorch (Pruning Tutorial)[https://pytorch.org/tutorials/intermediate/pruning_tutorial.html]
+
+        .. code-block:: python
+
+            parameters_to_prune = [
+                (model.mlp_1, "weight"),
+                (model.mlp_2, "weight")
+            ]
+
+            trainer = Trainer(
+                callbacks=[
+                    ModelPruning(
+                        'l1_unstructured',
```

Comment:
just to be consistent with the other being keyword args
```suggestion
                        pruning_fn='l1_unstructured',
```

Reply:
Done !